File: br-server/app/api/routes/training.py
"""培训课程列表 API 路由。"""

from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.database import get_db
from app.schemas.course import CourseListResponse, TrainingRoomListResponse
from app.services import training_service

router = APIRouter(prefix="/api/v1/training", tags=["training"])


@router.get("/rooms", response_model=TrainingRoomListResponse)
async def list_training_rooms(
    page: int = Query(1, ge=1),
    page_size: int = Query(10, ge=1),
    city_id: int | None = Query(None, ge=1),
    db: AsyncSession = Depends(get_db),
) -> TrainingRoomListResponse:
    return await training_service.list_training_rooms(
        db, page=page, page_size=page_size, city_id=city_id
    )


@router.get("/courses", response_model=CourseListResponse)
async def list_training_courses(
    page: int = Query(1, ge=1),
    page_size: int = Query(10, ge=1),
    category: str | None = Query(
        None,
        pattern="^(primaryschool|middleschool|postgraduate|civil_service|language|skills|professional)$",
    ),
    db: AsyncSession = Depends(get_db),
) -> CourseListResponse:
    return await training_service.list_courses(
        db, page=page, page_size=page_size, category=category
    )

File: br-app/src/api/training.js
import { get } from '@/utils/request'

/**
 * 获取培训室列表（含热门课程）
 * @param {Object} params - { page, page_size, city_id }
 */
export function getTrainingRooms(params) {
  return get('/api/v1/training/rooms', params)
}

/**
 * 获取培训课程列表
 * @param {Object} params - { page, page_size, category }
 */
export function getTrainingCourses(params) {
  return get('/api/v1/training/courses', params)
}

/**
 * 获取培训室详情（含课程列表和教师团队）
 * @param {number} roomId - 培训室ID
 */
export function getTrainingRoomDetail(roomId) {
  return get(`/api/v1/training/rooms/${roomId}`)
}
